src/FHIR/SMART/SmartLaunchController.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace OpenEMR\FHIR\SMART;
+
+use OpenEMR\Common\Auth\OpenIDConnect\Repositories\ClientRepository;
+use OpenEMR\Common\Uuid\UuidRegistry;
+use OpenEMR\Events\PatientDemographics\RenderEvent;
+use OpenEMR\RestControllers\AuthorizationController;
+use OpenEMR\Services\PatientService;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+
+// not sure I really like this here... it seems like some of this
+// should be encapsulated in a class that autoloading can reach.
+require_once(__DIR__ . '/../../../_rest_config.php');
+
+/**
+ * Class SmartLaunchController handles the display and launching of SMART apps from the user interface.
+ * @package OpenEMR\FHIR\SMART
+ */
+class SmartLaunchController
+{
+    /**
+     * @var EventDispatcher
+     */
+    private $dispatcher;
+
+    public function __construct(EventDispatcher $dispatcher = null)
+    {
+        $this->dispatcher = $dispatcher;
+    }
+
+    public function registerContextEvents()
+    {
+        $this->dispatcher->addListener(RenderEvent::EVENT_SECTION_LIST_RENDER_AFTER, [$this, 'renderPatientSmartLaunchSection']);
+    }
+
+    public function renderPatientSmartLaunchSection(RenderEvent $event)
+    {
+        $smartClients = $this->getSMARTClients();
+        // TODO: adunsulag we would filter the clients based on their smart capability & scopes they could send...
+        $pid = $event->getPid();
+        $patientService = new PatientService();
+        // going to work with string uuids
```

Comment:
in this case, we will need to ensure all pid's have a assigned uuid . The following line here will do that:
```
(new UuidRegistry(['table_name' => 'patient_data']))->createMissingUuids();
```

Reply:
I'm confused, shouldn't all patients already have a UUID if they've updated their OpenEMR install?  Or do we only create UUIDs on demand?

Reply:
It's a bit murky. Patient in theory should be covered, but wouldn't rely on it yet.

Reply:
@bradymiller Not sure if I need to specifically call you to ping you on this question, but I'm just wondering how the UUIDs are setup.

Reply:
check out the createMissingUuids() function to see details on the "shotgun approach" method. This function is used:
1. in upgrade for applicable tables (note plan to do these calls upgrade calls even for future upgrades)
2. when api makes calls to applicable tables

At the beginning of the uuid project, I did incorporate uuid creation for patient's in all the typical places a patient is created, but note it requires php to create the uuid (ie. database will not create a uuid automatically when adding an element (such as a new patient)), thus if a user imports patient's via a custom import script then it will not have uuids. So, to be safe for, whenever collecting items (when not grabbing a specific uuid) or going from the id/pid to uuid, best to ensure uuid's are populated.

Reply:
we can also go over it on the next call also (rarely are folks interested in this stuff, so would be very happy to discuss it :) )

Reply:
k, I'll add the code in.